samza-core/src/main/scala/org/apache/samza/coordinator/JobModelManager.scala

```diff
@@ -64,79 +63,36 @@ object JobModelManager extends Logging {
-   * b) Creates changeLogStream for task stores if it does not exists.
-   * c) Recomputes changelog partition mapping based on jobModel and job's configuration
-   * and writes it to the coordinator stream.
-   * d) Builds JobModelManager using the jobModel read from coordinator stream.
-   * @param coordinatorSystemConfig A config object that contains job.name
-   *                                job.id, and all system.&lt;job-coordinator-system-name&gt;.*Ch
-   *                                configuration. The method will use this config to read all configuration
-   *                                from the coordinator stream, and instantiate a JobModelManager.
+   * b) Recomputes changelog partition mapping based on jobModel and job's configuration.
+   * c) Builds JobModelManager using the jobModel read from coordinator stream.
+   * @param coordinatorStream
+   * @param changelogPartitionMapping
+   * @return JobModelManager
    */
-  def apply(coordinatorSystemConfig: Config, metricsRegistryMap: MetricsRegistryMap): JobModelManager = {
-    val coordinatorSystemConsumer: CoordinatorStreamSystemConsumer = new CoordinatorStreamSystemConsumer(coordinatorSystemConfig, metricsRegistryMap)
-    val coordinatorSystemProducer: CoordinatorStreamSystemProducer = new CoordinatorStreamSystemProducer(coordinatorSystemConfig, metricsRegistryMap)
-    info("Registering coordinator system stream consumer.")
-    coordinatorSystemConsumer.register
-    debug("Starting coordinator system stream consumer.")
-    coordinatorSystemConsumer.start
-    debug("Bootstrapping coordinator system stream consumer.")
-    coordinatorSystemConsumer.bootstrap
-    info("Registering coordinator system stream producer.")
-    coordinatorSystemProducer.register(SOURCE)
-
-    val config = coordinatorSystemConsumer.getConfig
-    info("Got config: %s" format config)
-    val changelogManager = new ChangelogPartitionManager(coordinatorSystemProducer, coordinatorSystemConsumer, SOURCE)
-    changelogManager.start()
-
+  def apply(coordinatorStream: CoordinatorStream, changelogPartitionMapping: util.Map[TaskName, Integer]) = {
+    val coordinatorSystemConsumer = coordinatorStream.getConsumer
+    val coordinatorSystemProducer = coordinatorStream.getProducer
     val localityManager = new LocalityManager(coordinatorSystemProducer, coordinatorSystemConsumer)
-    // We don't need to start() localityManager as they share the same instances with checkpoint and changelog managers.
-    // TODO: This code will go away with refactoring - SAMZA-678
 
-    localityManager.start()
+    val config = coordinatorSystemConsumer.getConfig
 
     // Map the name of each system to the corresponding SystemAdmin
     val systemAdmins = new SystemAdmins(config)
     val streamMetadataCache = new StreamMetadataCache(systemAdmins, 0)
-    val previousChangelogPartitionMapping = changelogManager.readChangeLogPartitionMapping()
 
     val processorList = new ListBuffer[String]()
```

Comment:
Not your code, but would be cleaner to use either: 
val x = (1 to containerCount).toList
val x = List.range(0, containerCount)

Reply:
Good idea. I'll clean that up.